src/core/EventDispatcher.js

```diff
@@ -76,6 +76,8 @@ Object.assign( EventDispatcher.prototype, {
 
 			}
 
+			event.target = null;
```

Comment:
Um, I'm not sure it's a good idea to alter the event object after it was dispatched. Users could work with the event object not just within the event listener. They could save the event and work with it later on. In this case it would be unexpected when `target` is suddenly `null`.

Reply:
I see your point, although anyone being called knows what target is (this), in the listener functions.  Is target used anywhere within Three?  I can't see any obvious uses. 

 An alternative would be to zap it specifically in the Object3D.remove() method after the dispatch call. In those cases anyone saving the event object could get surprises anyway since it would be shared with other _removed events after that point.

Reply:
I was thinking the same thing but if that's the case any reused event definitions like this would be problematic because `target` will get overwritten with every subsequent call. [TrackballControls](https://github.com/mrdoob/three.js/blob/dev/examples/jsm/controls/TrackballControls.js#L89-L91), [TransformControls](https://github.com/mrdoob/three.js/blob/dev/examples/jsm/controls/TransformControls.js#L69-L72), [OrbitControls](https://github.com/mrdoob/three.js/blob/dev/examples/jsm/controls/OrbitControls.js#L283-L285) and I'm sure others all use this same pattern.

These events were originally moved to be reused in order to save on memory allocation (see #17224, https://github.com/mrdoob/three.js/issues/17134#issuecomment-517128427) but I missed that the object was modified in the `dispatchEvent` function. I personally am okay with denoting that an event is only valid during the execution of the callback, which would allow us to save memory allocation in quite a few other places.

Setting the target to `null` in the `dispatchEvent` function would change the functionality relative to the built relative to the `EventTarget.dispatchEvent` function, though, so it might be more sensible to set target to null in the `remove()` function, instead.

Reply:
I totally forget that `three.js` code already reuse event object...

Anyway, in this case I don't have a real opinion on this. I prefer whatever @mrdoob likes. Also note that this class comes from a separate github repo. It could be useful to make changes also there: https://github.com/mrdoob/eventdispatcher.js/